addon/components/o-s-s/icon.ts

```diff
@@ -1,10 +1,11 @@
 import Component from '@glimmer/component';
 import { assert } from '@ember/debug';
+import { IconNames } from './iconName.enum';
 
 export type IconStyle = 'solid' | 'regular' | 'light' | 'duotone' | 'brand';
 
 interface OSSIconArgs {
-  icon: string;
+  icon: IconNames;
```

Comment:
Just typing it as a message so that other reviewers don't ask you.

It looks to me like you are missing changes in the icon.hbs file.

Currently if I use something like :
`<OSS::Icon @icon="Analytics" />`

The template will be rendered as :
`<i class="far Analytics"></i>` (with `far` being the default style value)

We need a way to convert "Analytics" back to "fa-analytics" so that font-awesome can match to its own name.

Reply:
UNLESS
You were thinking of using something to make that enum match directly the proper value in the template.
But then we'll need some other tool/library to handle this.

🤷